File: Exam_Portal/src/main/java/com/exam/service/QuizService.java
package com.exam.service;

import java.util.List;

import com.exam.model.exam.Category;
import com.exam.model.exam.Quiz;

public interface QuizService {

	public Quiz addQuiz(Quiz quiz);
	
	public Quiz updateQuiz(Quiz quiz);
	
	public Quiz updateByData(Quiz quiz);

	public List<Quiz> getQuizzes();
	
	public Quiz getQuiz(Long qid);

	public void deleteQuiz(Long qid);

	public List<Quiz> getQuizzesOfCategory(Category category);
	
	public List<Quiz> getActiveQuiz();

	public List<Quiz> getActiveQuizByCategory(Category category);


}

File: Exam_Portal_UI/src/app/service/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http : HttpClient) { }

  public quizzes(){
    return this.http.get(`${baseUrl}/quiz/`);
  }

  public addQuiz(quizData : any){
    return this.http.post(`${baseUrl}/quiz/`,quizData);
  }

  public deleteQuiz(qid : any){
    return this.http.delete(`${baseUrl}/quiz/${qid}`);
  }

  public getSingalQuiz(qid : any){
    return this.http.get(`${baseUrl}/quiz/${qid}`);
  }

  public updateQuiz(quizData : any){
    return this.http.put(`${baseUrl}/quiz/update`,quizData);
  }

  public getQuizzesOfCategory(qid : any){
    return this.http.get(`${baseUrl}/quiz/category/${qid}`);
  }

  public getAllActiveQuiz(){
    return this.http.get(`${baseUrl}/quiz/active`);
  }

  public getActiveQuizOfCategory(qid : any){
    return this.http.get(`${baseUrl}/quiz/category/active/${qid}`);
  }

}
